sky/task.py

```diff
@@ -68,6 +68,45 @@ def _is_valid_env_var(name: str) -> bool:
     return bool(re.fullmatch(_VALID_ENV_VAR_REGEX, name))
 
 
+def _fill_in_env_vars_in_storage_config(
+        storage_config: Dict[str, Any], task_envs: Dict[str,
+                                                        str]) -> Dict[str, Any]:
+    """Detects env vars in storage_config and fills them with task.envs.
+
+    This func tries to replace two fields in storage_config: 'source' and
+    'name'.
+
+    Env vars of the following forms are detected:
+        - ${ENV}
+        - $ENV
+    where <ENV> must appear in task.envs.
+
+    Example:
+        {'mode': 'MOUNT', 'name': '${GSBUCKET}', 'store': 'gcs'}
+        -> {mode: 'MOUNT', name: 'mybucket', store: 'gcs'}
+    """
+
+    # TODO(zongheng): support ${ENV:-default}?
+    def _substitute_env_vars(target: Union[str, List[str]],
+                             envs: Dict[str, str]) -> Union[str, List[str]]:
+        for key, value in envs.items():
+            pattern = r'\$\{?\b' + key + r'\b\}?'
```

Comment:
We may need to add a check for the key of the `task.envs` somewhere in our code, i.e. we need to make sure the `key ~ '[a-zA-Z][a-zA-Z0-9]*'`

Reply:
This will get checked by storage.py's validate_name().

Reply:
Ahh, I was trying to say that if the user provides a envvar like the following:
```
envs:
  hello world: some_value

run: |
  echo ${hello world}
```

The behavior will be different, if this env var is used in the `file_mounts` section.
```
file_mounts:
  /dst:
    source: ${hello world}
```

Reply:
Good call! Done.